### ISSUE TITLE
Custom Adblock Scriptlets are created in Default folder even if they are global

### ISSUE THREAD

**Issue by ghost** [brave/brave-browser#42894]
While Custom Scriptlets are global, they are saved in the Default Profile and not the root `User Data` folder as other global stuff like components or custom filters/lists (Local State file), while this doesn't affect any functionality, if you create multiple profiles, you are able to delete the first profile that was placed in the Default folder, and that means, when you remove it, the Default folder will stay in the User Data but only with the `AdBlock Custom Resources` folder.

So, even if you are not using the Default folder anymore, and now it is only the Profile # in place, if you create a scriptlet, a Default folder will be created with the `AdBlock Custom Resources` which means the behavior is not correct.

Everything works great, I even tested making Profile 1 as Default by editing the Local State file and it worked fine (once you remove a profile, the folder will not be reused and you can only do that by manually editing the Local State file).
But for example, if a user wants to 'reset' the Default profile, by just removing Default folder, now it will not be possible because you either make sure to remove everything in Default folder but `AdBlock Custom Resources` or the custom scriptlets will be removed.

It is great not to have to modify `resources.json` file anymore after years of doing it, but this seems to be the only weird thing I found about the new feature.

**Comment by Emi-Emi-Emi** [comment]
Also, I must say that there is no way to re-arrange the Custom Scriptlets, so rearranging means editing a Scriptlet and replaced it and then place the old one in whatever place you want, not the end of the world, it will just be slow to do it. But it would be nice to have a way to re-order the list.
The thing is, since the scriptlets are made in a LevelDB it's not possible to easily edit the files to do it 'manually', unlike other databases or the `resources.json` file.

And also, can I mention that I don't necessarily love the 'user-' prefix and you can only have lowercase names, but not the end of the world either, it works great so having to switch my custom scriptlets rules with 'new' names just to add the user- and match the lowercase was not exactly the an impossible task to do.

**Comment by Emi-Emi-Emi** [comment]
Also, when you remove scriptlets they don't get removed from UI until you refresh the page, while not a major issue, it is not a correct one.

**Comment by bsclifton** [comment]
cc: @antonok-edm @ryanbr

**Comment by boocmp** [comment]
>> Also, when you remove scriptlets they don't get removed from UI until you refresh the page, while not a major issue, it is not a correct one.

Hi, thanks, I really appreciate your feedback! Do you mean that scriplets remain in UI when you remove the folder from disk?

**Comment by Emi-Emi-Emi** [comment]
@boocmp no, it was when I removed the scriptlet with the trash bin icon in `brave://settings/shields/filters` it didn't go away until I refresh the settings page.

As you can see I talk in past tense, because apparently it works fine now. The only thing I did differently was to update Brave to `Nightly v1.75.102` earlier today, so I don't know if anything changed in .102 release, that somehow fixed the issue.

But I am deleting that comment since I can't reproduce it anymore. 👍

**Comment by boocmp** [comment]
I've landed the small fix yesterday about trash icon, so yes it should work fine now ! Thanks

**Comment by asihd0as8hd8shad** [comment]
I made different issues/feature requests for the other feedback, some little QoL improvements, the obvious request for Mobile support and some advanced topic that would be nice to have which is being able to assign a different mime to the scriptlet to function as a different thing for a redirection resource, since I used to modify the `resources.json` file to get my custom resources/scriptlets, I know the power and it would be amazing to treat any scriptlet as other types of resources like `resources.json`.

[Implement Custom Scriptlets in Brave Mobile](https://github.com/brave/brave-browser/issues/43105)

[Custom Scriptlets / Developer mode should have buttons to re-order the list of Scriptlets](https://github.com/brave/brave-browser/issues/43106)

[Custom Scriptlets names should allow uppercase characters and not be restricted to lowercase only.](https://github.com/brave/brave-browser/issues/43107)

[Allow Custom Scriptlets / Developer mode to assign different mime to the 'scriptlets'](https://github.com/brave/brave-browser/issues/43108)

**Comment by ShivanKaul** [comment]
Thanks a lot for trying it out and for raising issues! I'll close this issue out since it looks like it's fixed.
